src/lib.rs

```diff
@@ -673,18 +776,215 @@ decl_module! {
+
+            // Generate event
+            Self::deposit_event(RawEvent::CategoryUpdated(category_id, archived_change_to, deleted_changed_to));
+
+            Ok(())
+        }
+
+        /// Create new thread in category
+        fn create_thread(origin, category_id: CategoryId, title: Vec<u8>, text: Vec<u8>) -> dispatch::Result {
+
+            /*
+             * Update SPEC with new errors,
+             * and mutation of Category class,
+             * as well as side effect to update Category::num_threads_ever_created.
+             */ 
+
+            // Check that its a valid signature
+            let who = ensure_signed(origin)?;
+
+            // Check that account is forum member
+            Self::ensure_is_forum_member(&who)?;
+
+            // Get path from parent to root of category tree.
+            let category_tree_path = Self::ensure_valid_category_and_build_category_tree_path(category_id)?;
+
+            // No ancestor is blocking us doing mutation in this category
+            Self::ensure_can_mutate_in_path_leaf(&category_tree_path)?;
+            
+            // Validate title
+            ensure_thread_title_is_valid(&title)?;
+
+            // Validate post text
+            ensure_post_text_is_valid(&text)?;
+
+            /*
+             * Here it is safe to mutate state.
+             */
+
+            // Get category
+            let category = category_tree_path.first().unwrap().clone();
+
+            assert_eq!(category.id, category_id);
+            
+            // Update (unmoderated) thread count in corresponding category
+            <CategoryById<T>>::mutate(category_id, |c| {
+                c.num_direct_unmoderated_threads += 1;
+            });
+
+            // Get thread counter
+            let new_thread_id = <NextThreadId<T>>::get();
+
+            // Create and add new thread
+            let new_thread = Thread {
+                id : new_thread_id,
+                title : title.clone(),
+                category_id: category_id,
+                thread_nr_in_category: category.num_threads_ever_created(),
+                moderation : None,
+                num_unmoderated_posts: 0,
+                num_moderated_posts: 1,
+                created_at : Self::current_block_and_time(),
+                author_id : who.clone()
+            };
+
+            <ThreadById<T>>::insert(new_thread_id, new_thread);
+
+            // Update next thread id
+            <NextThreadId<T>>::mutate(|n| {
+                *n += 1;
+            });
+
+            // Make and add initial post
+
+            // TODO: perhaps factor out later?
+
+            let new_post_id = <NextPostId<T>>::get();
+
+            <NextPostId<T>>::put(new_post_id + 1);
+
+            // Create and add new post
+            let new_post = Post {
+                id: new_post_id,
+                thread_id: new_thread_id,
+                post_nr_in_thread: 1, // Starts at 1
```

Comment:
Would you like to create constant for a default value of any entity in a list and use it for subcategories, threads, posts?
```suggestion
// somewhere above:
const DEFAULT_POSITION_IN_LIST: u32 = 1;
// ... other code...

                post_nr_in_thread: DEFAULT_POSITION_IN_LIST, // Starts at 1
```

Reply:
This issue is resolved in https://github.com/Joystream/substrate-forum-module/pull/13 , so there is no longer a constant.